gordo_components/workflow/config_elements/machine.py

```diff
@@ -73,9 +73,10 @@ def from_config(  # type: ignore
         local_runtime = config.get("runtime", dict())
         runtime = patch_dict(config_globals.get("runtime", dict()), local_runtime)
 
-        dataset = Dataset.from_config(
-            config.get("dataset") or config_globals.get("dataset")
+        dataset_config = patch_dict(
+            config.get("dataset", dict()), config_globals.get("dataset", dict())
         )
+        dataset = Dataset.from_config(dataset_config)
```

Comment:
I like it!

Reply:
Awesome! I'll just do a test on the cluster to see that I didn't shoot myself in the foot before merging :-)